tests/databaseDiff.js

```diff
@@ -0,0 +1,253 @@
+const ERR = require('async-stacktrace');
+const async = require('async');
+const fs = require('fs-extra');
+const path = require('path');
+const pg = require('pg');
+const pgArray = require('pg').types.arrayParser;
+const assert = require('chai').assert;
+const colors = require('colors');
+const _ = require('lodash');
+const jsdiff = require('diff');
+
+const sqldb = require('../lib/sqldb');
+const sqlLoader = require('../lib/sql-loader');
+const describe = require('../tests/databaseDescribe');
+
+module.exports = {};
+
+function diff(db1, db2, options, callback) {
+    function formatText(text, formatter) {
+        if (options.coloredOutput && formatter) {
+            return formatter(text);
+        }
+        return text;
+    }
+
+    let db1Name = db1.type === 'database' ? db1.name : db1.path;
+    let db2Name = db2.type === 'database' ? db2.name : db2.path;
+
+    let description1, description2;
+    let result = '';
+
+    async.series([
+        (callback) => {
+            loadDescription(db1, (err, description) => {
+                if (ERR(err, callback)) return;
+                description1 = description;
+                callback(null);
+            });
+        },
+        (callback) => {
+            loadDescription(db2, (err, description) => {
+                if (ERR(err, callback)) return;
+                description2 = description;
+                callback(null);
+            });
+        },
+        (callback) => {
+            // Determine if both databases have the same tables
+            let missingFrom1 = _.difference(_.keys(description2.tables), _.keys(description1.tables));
+            let missingFrom2 = _.difference(_.keys(description1.tables), _.keys(description2.tables));
+
+            if (missingFrom1.length > 0) {
+                let db1NameBold = formatText(db1Name, colors.bold);
+                result += formatText(`Tables missing from ${db1NameBold} (${db1.type})\n`, colors.underline);
+                result += missingFrom1.map(table => `    ${table}`).join('\n') + '\n\n';
```

Comment:
Should these be colored green with a leading + on the line? I think they are things that should be added to DB1? Similarly for the next list of tables that could be colored red. I find that when I look at pg_diff output it's easy to overlook these lists because I'm scanning for red/green or -/+ as indicators of differences.